Documentation/diagnostics/source_link.md

```diff
@@ -0,0 +1,103 @@
+# Source Link
+[Portable PDB](https://github.com/dotnet/core/blob/master/Documentation/diagnostics/portable_pdb.md) introduces a new developer productivity feature called Source Link. Source Link allows a Portable PDB to embed unique information about the assembly's original source code during compilation. The [Roslyn](https://github.com/dotnet/roslyn/wiki/Roslyn%20Overview) compiler flag for Source Link is `/sourcelink:<file>`.
+
+## Why Source Link?
+Most debugging is done against locally built source code on a developer's machine. In this scenario, matching the binary to the source is not difficult. However, there are many debugging scenarios where the original source code is not immediately availble. Two good examples of this are debugging crash dumps or third party libraries. In these scenarios, it can be very difficult for a developer to acquire the exact source code that was built to produce the binary they are debugging. Source Link solves this problem by embedding unique information about the source code (such as a git commit hash) in the Portable PDB. Diagnostic tools, such as debuggers, can use this unique information to retrieve the original source code from hosted services (such as [Github](https://github.com)).
+
+## MSBuild Task
+TBD
+### Example
+`TBD`
+
+## Source Link File Specification
+The `/sourcelink:<file>` compiler flag will embed a JSON configuration file in the Portable PDB. This configuration file would be generated as part of the build process. A default MSBuild task is already availble for Source Link generation (see above), but the generation can be customized to suit the needs of the project. The JSON configuration file contains a simple mapping of local file path to URL where the source file can be retrieved via http or https. A debugger would retrieve the originfal file path of the current location from the Portable PDB, look this path up in the Source Link map, and use the resulting URL to download the source file.
+
+### Source Link JSON Schema
+```json
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "title": "SourceLink",
+    "description": "A mapping of source file paths to URI's",
+    "type": "object",
+    "properties": {
+        "documents": {
+            "type": "object",
+            "minProperties": 1,
+            "additionalProperties": {
+                "type": "string"
+            },
+            "descripttion": "Each document is defined by a file path and a URI. Original source file paths are compared 
+                             case insensitive to documents and the resulting URI is used to download source. The document 
+                             may contain an asterisk to represent a wildcard in order to match anything in the asterisk's 
+                             location. If the document contains an asterisk, the URI must also contain an aterisk. The 
```

Comment:
I will clarify this, the exact rules are these:
1. The only acceptable wildcard is one and only one '*', which if present will be replaced by a relative path
2. If the filepath does not contain a *, the uri cannot contain a * and if the filepath contains a * the uri must contain a *
3. If the filepath contains a *, it must be the final character
4. If the uri contains a *, it may be anwhere in the uri